src/Microsoft.AspNetCore.Mvc.ApiExplorer/ProblemAttributeApiDescriptionProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    public class ProblemAttributeApiDescriptionProvider : IApiDescriptionProvider
+    {
+        private readonly IModelMetadataProvider _modelMetadaProvider;
+
+        public ProblemAttributeApiDescriptionProvider(IModelMetadataProvider modelMetadataProvider)
+        {
+            _modelMetadaProvider = modelMetadataProvider;
+        }
+
+        /// <remarks>
+        /// The order is set to execute after the <see cref="DefaultApiDescriptionProvider"/>.
+        /// </remarks>
+        public int Order => -1000 + 10;
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        public void OnProvidersExecuting(ApiDescriptionProviderContext context)
+        {
+            foreach (var apiDescription in context.Results)
+            {
+                if (!apiDescription.ActionDescriptor.Properties.ContainsKey(ProblemDescriptionApplicationModelProvider.ProblemDescriptionAttributeKey))
+                {
+                    continue;
+                }
+
+                var parameters = apiDescription.ActionDescriptor.Parameters.Concat(apiDescription.ActionDescriptor.BoundProperties);
+                if (parameters.Any())
+                {
+                    apiDescription.SupportedResponseTypes.Add(CreateProblemResponse(StatusCodes.Status400BadRequest));
+
+                    if (parameters.Any(p => string.Equals("id", p.Name, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Would we want to also look for things of the form `blahId`, i.e.., `EndsWith("Id", StringComparison.Ordinal)`? I personally always tend to name things `customerId`, `productId`, etc.

Also this is kind of English-language centric. How much precidence is there for giving special meaning to `id` elsewhere? I seem to recall that EF has some conventions around this. But if I'm wrong then maybe the `id` property name/suffix should be configurable.

Reply:
Good idea on the EF bit. I'll take a gander at their code.

Reply:
Agreed, I think we can do ahead and do `EndsWith("Id"...)` right now. 

I'm not worried about extensibility for this class right now, we need to do it but we need to get to a working E2E so we can start on the other parts of this.

Reply:
+1 for ```EndsWith("Id"...)```. Extensibility, possibly in the form of an action parameter attribute, to indicate "this parameter can trigger a 404" would be useful.